solutions/devsprint-6/OnboardingChallenge/Screens/Components/ButtonView.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  ButtonView.swift
+//  OnboardingChallenge
+//
+//  Created by Gustavo Soares on 12/11/21.
+//
+
+import UIKit
+
+struct ButtonViewConfiguration {
+    var title: String
+    var action: UIAction
+}
+
+class ButtonView: UIView {
+    
+    private(set) var buttonWidth: CGFloat = 300
+    private(set) var buttonHeight: CGFloat = 60
+    
+    private lazy var button: UIButton = {
+        let button = UIButton()
+        button.backgroundColor = .blue
+        button.titleLabel?.font = UIFont.boldSystemFont(ofSize: 18)
+        button.tintColor = .white
+        button.translatesAutoresizingMaskIntoConstraints = false
+        return button
+    }()
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    init() {
+        super.init(frame: .zero)
+        setupViews()
+    }
+    
+    func updateView(with buttonViewConfiguration: ButtonViewConfiguration) {
+        button.setTitle(buttonViewConfiguration.title, for: .normal)
+        button.addAction(buttonViewConfiguration.action, for: .touchUpInside)
+    }
+}
+
+extension ButtonView: ViewCode {
+    func configureSubViews() {
+        addSubview(button)
+    }
+    
+    func configureSubviewsConstraints() {
+        NSLayoutConstraint.activate([
+            button.widthAnchor.constraint(equalToConstant: buttonWidth),
```

Comment:
Você poderia trocar essa `widthAnchor` por duas outras constraints: `leadingAnchor` e `trailingAnchor`. Assim, o componente ButtonView vai ter um comprimento que pode variar de acordo com a tela em que ele for utilizado.